packages/nuxt/src/core/plugins/dev-only.ts

```diff
@@ -9,7 +9,7 @@ interface DevOnlyPluginOptions {
 }
 
 export const DevOnlyPlugin = createUnplugin((options: DevOnlyPluginOptions) => {
-  const DEVONLY_COMP_RE = /<(dev-only|DevOnly)>[\s\S]*?<\/(dev-only|DevOnly)>/g
+  const DEVONLY_COMP_RE = /<(?:dev-only|DevOnly)>[^<]*(?:<template\s+#fallback>(?<fallback>[\s\S]*?)<\/template>)?[\s\S]*?<\/(?:dev-only|DevOnly)>/g
```

Comment:
hmmm.... i forgot to submit the review

Reply:
thanks for looking at this. we should probably add the handling to `TreeShakeTemplatePlugin` ...

Reply:
Thanks for sharing! I was chaining the two regex/matches earlier but putting it into one needed that `[^<]` which caused it

Reply:
How about using `ultrahtml` instead of the regex and keep the `dev-only` plugin ? `TreeShakeTemplatePlugin` won't remove `ssrRender(DevOnly...)` but only remove its slots that are not `fallback` or `placeholder`.

Reply:
Completely up for using parsers over regex for safer template handling! Should it wait for #20284 as it adds `ultrahtml` to Nuxt dependencies first?

Reply:
you can do it now, it won't cause any complex merge conflict 🙂

Reply:
Currently gone for splitting regex (in #20840) as `ultrahtml` could cause trade-off with performance (through safety).